pkg/operator/ceph/disruption/clusterdisruption/reconcile.go

```diff
@@ -111,8 +114,19 @@ func (r *ReconcileClusterDisruption) reconcile(request reconcile.Request) (recon
 		// feature disabled for this cluster. not requeueing
 		return reconcile.Result{Requeue: false}, nil
 	}
-	//signal to the nodedrain controller to start
-	r.context.ReconcileCanaries.Update(true)
+
+	// delete any legacy blocking pdbs for osd
+	err := r.deleteLegacyPDBForOSD()
```

Comment:
Deleting legacy PDBs and canary pods doesn't need to be done every reconcile. Is there somewhere we can do it once at controller startup? Or else we could have a package var that keeps track if it's already done. If it's a rare reconcile this is fine, but this reconcile is potentially every 30s, right?

Reply:
using package level variable.  Legacy resources will be deleted only on the first reconcile.